text/0000-try-expr.md

```diff
@@ -0,0 +1,834 @@
+    - **Used as crate?** [*No*](https://crates.io/crates/catch).
+    - **Usage (sourcegraph):** **21** regex:
+    ```
+    repogroup:crates case:yes max:400
+    \b((let|const|type|)\s+catch\s+=|(fn|impl|mod|struct|enum|union|trait)\s+catch)\b
+    ```
+7. **Consistency with old learning material:** Untaught
+
+### Review
+
+We believe `catch` to be a poor choice of keyword, because it:
+
+1. is used nowhere in other languages to demarcate the body which can result in
+   an exceptional path. Instead, it is almost exclusively used for exception
+   handlers of the form: `catch(pat) { recover_expr }`.
+4. extending `catch` with handlers will require a different word such as
+   `handler` to get `catch { .. } handler(e) { .. }` semantics if we want.
+   This inversion compared to a lot of other languages will only harm
+   teachability of the language and steal a lot of our strangeness budget.
+2. it is less brief than `try`.
+3. the consistency wrt. methods in the standard library is low -
+   there's only `catch_unwind`, but that has to do with panics,
+   not `Try` style exceptions.
+
+## Alternative: keeping `do catch { .. }`
+
+1. **Fidelity to the construct's actual behavior:** High
+2. **Precedent from existing languages:**
+    + `do`: Haskell, Idris
+    + `catch`: None, see [prior-art]
+3. **Brevity / Length:** 8
+4. **Consistency with the naming of the trait used for `?`:** Inconsistent
+5. **Consistency with related libstd fn conventions:** Tiny bit consistent
+6. **Risk of breakage:** Impossible (already reserved keyword)
+    - **Used in std:** *No*, the form `$ident $ident` is not a legal identifier.
+    - **Used as crate?** *No*, as above.
+    - **Usage (sourcegraph):** **0** regex: N/A
+7. **Consistency with old learning material:** Untaught
+
+An alternative would be to simply use the `do catch { ... }` syntax we have
+in the nightly compiler. However, this syntax was not in the accepted `catch`
+RFC and was only a temporarly fix around `catch { .. }` not working.
+
+## Alternative: `do try { .. }`
+
+1. **Fidelity to the construct's actual behavior:** High
+2. **Precedent from existing languages:**
+    + `do`: Haskell, Idris
+    + `try`: A lot, see [prior-art]
+    1. **Popularity of the languages:** Massive accumulated dominance
+    2. **Fidelity to behavior in those languages:** High
+    3. **Familiarity with respect to their analogous constructs:** High
+3. **Brevity / Length:** 6 (including space)
+4. **Consistency with the naming of the trait used for `?`:** Moderately consistent
+5. **Consistency with related libstd fn conventions:** Moderately consistent
+6. **Risk of breakage:** Impossible (already reserved keyword)
+    - **Used in std:** *No*, the form `$ident $ident` is not a legal identifier.
+    - **Used as crate?** *No*, as above.
+    - **Usage (sourcegraph):** **0** regex: N/A
+7. **Consistency with old learning material:** Untaught
+
```

Comment:
You're missing an alternative here -- a completely new keyword. Could be `result`. That has a higher chance of breakage but it's edition'd so maybe not an issue? I prefer catch over try but I strongly prefer something new over both.

Reply:
I see trap, fallible, collect, capture, etc all being considered later in the RFC.

Reply:
I've now added a review of the `result` keyword below `trap`.